Sources/XCTQueues/TestQueueDriver.swift

```diff
@@ -0,0 +1,88 @@
+import Queues
+import Vapor
+
+extension Application.Queues.Provider {
+    public static var test: Self {
+        .init {
+            $0.queues.use(custom: TestQueuesDriver())
+        }
+    }
+}
+
+struct TestQueuesDriver: QueuesDriver {
+    public func makeQueue(with context: QueueContext) -> Queue {
+        TestQueue(context: context)
+    }
+    
+    public func shutdown() {
+        // nothing
+    }
+}
+
+extension Application.Queues {
+    public final class TestQueueStorage {
+        public var jobs: [JobIdentifier: JobData] = [:]
+        public var queue: [JobIdentifier] = []
+        
+        /// Returns the first job in the queue of the specific `J` type.
+        public func first<J>(_ job: J.Type) -> J.Payload?
+            where J: Job {
```

Comment:
spacing nit, other put the `{` on a newline and don't indent the body of the func
```swift
func foo<T> 
    where T: Bar
{ ... }
```